src/app/files/file-explorer.js

```diff
@@ -276,6 +316,217 @@ fileExplorer.prototype.init = function () {
   return this.container
 }
 
+fileExplorer.prototype.publishToGist = function (fileProviderName) {
+  modalDialogCustom.confirm(
+    null,
+    'Are you sure you want to publish all your files anonymously as a public gist on github.com?',
+    () => { this.toGist(fileProviderName) }
+  )
+}
+
+fileExplorer.prototype.uploadFile = function (event) {
```

Comment:
One thing that should perhaps be done in other PR is to move these functions to another file